### DISCUSSION TITLE
How to send a signed message with Local Account

### DISCUSSION THREAD

**Discussion by thejasondt** [wevm/viem#617] [Question] [answered]
Hey!

I have recently moved over from Ethers to viem and am struggling to `writeContract` from Local Account with viem and Infura.

The ethers code is pretty simple with `const signer = wallet.connect(provider)` (`const wallet = new Wallet(privateKey, provider)`
that we pass to the `new Contract` before calling `contract.enterFunction`

There isn't an equivalent in the migration guide of creating a signer from the wallet as I do in Ethers.

When I try `const hash = await walletClient.writeContract(sim)` in viem I get:

"Server error TransactionExecutionError: The method does not exist / is not available.\n\nURL: wss://sepolia.infura.io/ws/v3/apiKey\nRequest body: {\"method\":\"eth_sendTransaction\",\"params\":[{\"data\":\"\",\"from\":\"\",\"to\":\"\"}]}\n \nRequest Arguments:\n  from:  \n  to:    \n  data:  \n\nDetails: The method eth_sendTransaction does not exist/is not available\nVersion: viem@0.3.35"

Is Ethers using `eth_sendRawTransaction` under the hood as opposed to `eth_sendTransaction`

How can I send a signed message/send raw with writeContract or another way around this?

**Comment by jxom** [comment]
Do you have your Local Account attached to the `walletClient`? https://viem.sh/docs/clients/wallet.html#optional-hoist-the-account-1

**Reply by thejasondt** [reply]
Yep,

`const account = privateKeyToAccount('0x' + privateKey)
        const walletClient = createWalletClient({
            account,
            chain,
            transport
          });
`

Do I need to add the Local account to PublicClient also when doing the sim? EDIT: no, it does no accept account as a var

I am adding the `{ request }` from `simulateContract` to `writeContract`

**Reply by tmm** [reply]
Can you pop a link to a minimal reproduction? Can use [new.viem.sh](https://new.viem.sh) as a starting point.

**Reply by thejasondt** [reply, marked as answer]
I found the issue.

I needed to simulateContract with the Local Account also - perhaps it would be good to hoist to publicClient as with the walletClient
